### DISCUSSION TITLE
Slice a triangle mesh with a plane，output a 2D triangle mesh

### DISCUSSION THREAD

**Discussion by Lafinger** [libigl/libigl#2081] [Q&A]
How can I get a 2D triangle mesh by slicing a triangle mesh with a plane?

**Comment by alecjacobson** [comment]
You can do this with igl::copyleft::cgal::intersect_with_half_space

**Reply by Lafinger** [reply]
Thanks!

**Reply by Lafinger** [reply]
I try to use it, but throw out an error. I want to know how to use "igl::copyleft::cgal::intersect_with_half_space()".

Here's how I use it:
#include <igl/readOFF.h>
#include <igl/opengl/glfw/Viewer.h>
#include <igl/copyleft/cgal/intersect_with_half_space.h>

Eigen::MatrixXd V;
Eigen::MatrixXi F;
Eigen::MatrixXd VC;
Eigen::MatrixXi FC;
Eigen::MatrixXi JC;

int main(int argc, char *argv[])
{
  // Load a mesh in OFF format
  igl::readOFF("./mouseBrain.off", V, F);
  typedef CGAL::Plane_3<CGAL::Epeck> Plane;
  typedef CGAL::Point_3<CGAL::Epeck> Point;
  typedef CGAL::Vector_3<CGAL::Epeck> Vector;
  Plane P(Point(0.0, 200.0, 0.0), Vector(0.0, 1.0, 0.0));
  igl::copyleft::cgal::intersect_with_half_space(V, F, P, VC, FC, JC);
  std::cout << VC << std::endl; // **VC is empty**

  //// Plot the mesh
  //igl::opengl::glfw::Viewer viewer;
  //viewer.data().set_mesh(VC, FC);
  //viewer.launch();
}

**Reply by alecjacobson** [reply]
This is working for me:
```
#include <igl/read_triangle_mesh.h>
#include <igl/opengl/glfw/Viewer.h>
#include <igl/copyleft/cgal/intersect_with_half_space.h>

Eigen::MatrixXd V;
Eigen::MatrixXi F;
Eigen::MatrixXd VC;
Eigen::MatrixXi FC;
Eigen::MatrixXi JC;

int main(int argc, char *argv[])
{
  igl::read_triangle_mesh(argv[1], V, F);
  typedef CGAL::Plane_3<CGAL::Epeck> Plane;
  typedef CGAL::Point_3<CGAL::Epeck> Point;
  typedef CGAL::Vector_3<CGAL::Epeck> Vector;
  Plane P(
    Point(
      V.col(0).array().mean(),
      V.col(1).array().mean(),
      V.col(2).array().mean()),
    Vector(0.0, 1.0, 0.0));
  igl::copyleft::cgal::intersect_with_half_space(V, F, P, VC, FC, JC);
  //std::cout << VC << std::endl; // VC is empty
  
  // Plot the mesh

  igl::opengl::glfw::Viewer viewer;
  viewer.data().set_mesh(VC, FC);
  viewer.append_mesh();
  viewer.data().set_mesh(V, F);
  viewer.launch();
}
```

![Screen Shot 2022-10-03 at 1 28 17 PM](https://user-images.githubusercontent.com/2241689/193640959-48fffa91-f74a-486f-898d-f625e1de9284.png)
![Screen Shot 2022-10-03 at 1 28 12 PM](https://user-images.githubusercontent.com/2241689/193640963-373f7a13-ad1c-43bb-baf0-a973cb33a713.png)

Maybe your plane just doesn't intersect your model?
